--- postfix-2.1.5/conf/main.cf.aliases	2004-10-18 13:57:18.932558373 +0200
+++ postfix-2.1.5/conf/main.cf	2004-10-18 13:57:27.162425581 +0200
@@ -384,7 +384,7 @@
 #
 #alias_database = dbm:/etc/aliases
 #alias_database = dbm:/etc/mail/aliases
-#alias_database = hash:/etc/aliases
+alias_database = hash:/etc/aliases
 #alias_database = hash:/etc/aliases, hash:/opt/majordomo/aliases
 
 # ADDRESS EXTENSIONS (e.g., user+foo)
