diff -up postfix-2.5.1/conf/main.cf.config postfix-2.5.1/conf/main.cf
--- postfix-2.5.1/conf/main.cf.config	2007-12-18 08:50:25.000000000 -0500
+++ postfix-2.5.1/conf/main.cf	2008-08-28 17:22:21.000000000 -0400
@@ -113,6 +113,7 @@ mail_owner = postfix
 #inet_interfaces = all
 #inet_interfaces = $myhostname
 #inet_interfaces = $myhostname, localhost
+inet_interfaces = localhost
 
 # The proxy_interfaces parameter specifies the network interface
 # addresses that this mail system receives mail on by way of a
@@ -157,7 +158,7 @@ mail_owner = postfix
 #
 # See also below, section "REJECTING MAIL FOR UNKNOWN LOCAL USERS".
 #
-#mydestination = $myhostname, localhost.$mydomain, localhost
+mydestination = $myhostname, localhost.$mydomain, localhost
 #mydestination = $myhostname, localhost.$mydomain, localhost, $mydomain
 #mydestination = $myhostname, localhost.$mydomain, localhost, $mydomain,
 #	mail.$mydomain, www.$mydomain, ftp.$mydomain
@@ -379,7 +380,7 @@ unknown_local_recipient_reject_code = 55
 # "postfix reload" to eliminate the delay.
 #
 #alias_maps = dbm:/etc/aliases
-#alias_maps = hash:/etc/aliases
+alias_maps = hash:/etc/aliases
 #alias_maps = hash:/etc/aliases, nis:mail.aliases
 #alias_maps = netinfo:/aliases
 
diff -up postfix-2.5.1/conf/postfix-files.config postfix-2.5.1/conf/postfix-files
--- postfix-2.5.1/conf/postfix-files.config	2008-01-23 19:13:49.000000000 -0500
+++ postfix-2.5.1/conf/postfix-files	2008-08-28 17:24:37.000000000 -0400
@@ -128,8 +128,8 @@ $config_directory/postfix-script:f:root:
 $config_directory/postfix-script-sgid:f:root:-:755:o
 $config_directory/postfix-script-nosgid:f:root:-:755:o
 $config_directory/post-install:f:root:-:755
-$manpage_directory/man1/mailq.1:f:root:-:644
-$manpage_directory/man1/newaliases.1:f:root:-:644
+$manpage_directory/man1/mailq.postfix.1:f:root:-:644
+$manpage_directory/man1/newaliases.postfix.1:f:root:-:644
 $manpage_directory/man1/postalias.1:f:root:-:644
 $manpage_directory/man1/postcat.1:f:root:-:644
 $manpage_directory/man1/postconf.1:f:root:-:644
@@ -141,9 +141,9 @@ $manpage_directory/man1/postlog.1:f:root
 $manpage_directory/man1/postmap.1:f:root:-:644
 $manpage_directory/man1/postqueue.1:f:root:-:644
 $manpage_directory/man1/postsuper.1:f:root:-:644
-$manpage_directory/man1/sendmail.1:f:root:-:644
+$manpage_directory/man1/sendmail.postfix.1:f:root:-:644
 $manpage_directory/man5/access.5:f:root:-:644
-$manpage_directory/man5/aliases.5:f:root:-:644
+$manpage_directory/man5/aliases.postfix.5:f:root:-:644
 $manpage_directory/man5/body_checks.5:f:root:-:644
 $manpage_directory/man5/bounce.5:f:root:-:644
 $manpage_directory/man5/canonical.5:f:root:-:644
