diff --git a/postfix-install b/postfix-install
index 922402b..9cc52d5 100644
--- a/postfix-install
+++ b/postfix-install
@@ -40,6 +40,10 @@
 #	is installed from a pre-built package or from source code.
 #
 #	Arguments
+# .IP -keep-new-mtime
+#	When installing files preserve new file's mtime timestamps.
+#	Otherwise, mtimes will be set to the time that postfix-install
+#	is run.
 # .IP -non-interactive
 #	Do not ask the user for parameter settings. Installation parameters
 #	are specified via one of the non-interactive methods described
@@ -210,6 +214,7 @@ if [ -n "$SHLIB_ENV_VAR" ]; then
 fi
 
 USAGE="Usage: $0 [name=value] [option]
+    -keep-new-mtime         Preserve new file's mtime timestamps.
     -non-interactive        Do not ask for installation parameters.
     -package                Build a ready-to-install package.
     name=value              Specify an installation parameter".
@@ -225,6 +230,8 @@ do
 	*=*) IFS= eval $arg; IFS="$BACKUP_IFS";;
   -non-int*) non_interactive=1;;
    -package) need_install_root=install_root;;
+-keep-new-mtime)
+             keep_new_mtime=1;;
 	  *) echo "$0: Error: $USAGE" 1>&2; exit 1;;
     esac
     shift
@@ -305,6 +312,7 @@ compare_or_replace() {
 	echo Updating $dst...
 	rm -f $tempdir/junk || exit 1
 	cp $src $tempdir/junk || exit 1
+	test -z "$keep_new_mtime" || touch -r $src $tempdir/junk || exit 1
 	mv -f $tempdir/junk $dst || exit 1
 	test -z "$owner" || chown $owner $dst || exit 1
 	test -z "$group" || chgrp $group $dst || exit 1
diff --git a/Makefile.in b/Makefile.in
index 6d9d9b4..ae61297 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -122,23 +122,24 @@ printfck: update
 install: update
 	SHLIB_ENV_VAR= SHLIB_ENV_VAL= \
 	$(SHLIB_ENV) shlib_directory=$(SHLIB_DIR_OVERRIDE) $(SHELL) \
-		postfix-install
+		postfix-install $(POSTFIX_INSTALL_OPTS)
 
 package: update
 	SHLIB_ENV_VAR= SHLIB_ENV_VAL= \
 	$(SHLIB_ENV) shlib_directory=$(SHLIB_DIR_OVERRIDE) $(SHELL) \
-		postfix-install -package
+		postfix-install -package $(POSTFIX_INSTALL_OPTS)
 
 upgrade: update
 	SHLIB_ENV_VAR= SHLIB_ENV_VAL= \
 	$(SHLIB_ENV) shlib_directory=$(SHLIB_DIR_OVERRIDE) $(SHELL) \
-		postfix-install -non-interactive
+		postfix-install -non-interactive $(POSTFIX_INSTALL_OPTS)
 	
 
 non-interactive-package: update
 	SHLIB_ENV_VAR= SHLIB_ENV_VAL= \
 	$(SHLIB_ENV) shlib_directory=$(SHLIB_DIR_OVERRIDE) $(SHELL) \
-		postfix-install -non-interactive -package
+		postfix-install -non-interactive -package \
+		$(POSTFIX_INSTALL_OPTS)
 
 depend clean:
 	set -e; for i in $(DIRS); do \
diff --git a/makedefs b/makedefs
index 8f78601..eb11838 100644
--- a/makedefs
+++ b/makedefs
@@ -1196,5 +1196,6 @@ PLUGIN_MAP_OBJ_UPDATE = $PLUGIN_MAP_OBJ_UPDATE
 PLUGIN_MAP_SO_MAKE = $PLUGIN_MAP_SO_MAKE
 PLUGIN_MAP_SO_UPDATE = $PLUGIN_MAP_SO_UPDATE
 PLUGIN_LD = $PLUGIN_LD
+POSTFIX_INSTALL_OPTS = $POSTFIX_INSTALL_OPTS
 # Application-specific rules.
 EOF
